vertica_python/vertica/connection.py

```diff
@@ -440,25 +445,31 @@ def enable_ssl(self, raw_socket, ssl_options):
         return raw_socket
 
     def establish_connection(self):
-        addr = self.address_list.peek()
+        addrinfo = self.address_list.peek()
         raw_socket = None
         last_exception = None
 
         # Failover: loop to try all addresses
-        while addr:
+        while addrinfo:
+            (family, socktype, proto, canonname, sockaddr) = addrinfo
             last_exception = None
-            host, port = addr
+
+            # _AddressList filters all addrs to AF_INET and AF_INET6, which both
+            # have host and port as values 0, 1 in the sockaddr tuple.
+            host = sockaddr[0]
+            port = sockaddr[1]
```

Comment:
Line 406 `socket_host, socket_port = raw_socket.getpeername()` should make the similar change.

`raw_socket.getpeername()` returns a `(address, port)` 2-tuple for AF_INET, and a `(address, port, flow info, scope id)` 4-tuple for AF_INET6. So for __AF_INET6__, `socket_host, socket_port = raw_socket.getpeername()` will raise an error: ValueError: too many values to unpack (expected 2)